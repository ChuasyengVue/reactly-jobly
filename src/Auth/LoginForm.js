import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

const LoginForm = ({ login }) => {
    const navigate = useNavigate();

    const[formData, setFormData] = useState({
        username:"",
        password:""
    });

    const handleChange = (evt) =>{
        const {name, value} = evt.target;
        setFormData(formData => ({...formData, [name]:value }));
    }

    const handleSubmit = async (evt) =>{
        evt.prevenDefault();
        try {
            await login(formData)
            navigate('/');
        } catch (error) {
            console.error("Login failed:", error);
        }
    }

    return(
        <div>
            <form onSubmit={handleSubmit}>
                <h3>Log In</h3>
                <div>
                    <label>Username</label>
                    <input 
                    name="username"
                    value={formData.username}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Password</label>
                    <input
                    name="password"
                    type="password"
                    value={formData.password}
                    onChange={handleChange}
                    required
                    />
                </div>

                <button type='submit'> Submit </button>
            </form>
        </div>
    )

}

export default LoginForm;