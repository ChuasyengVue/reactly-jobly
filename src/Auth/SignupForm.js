import React, { useState } from "react";
import { useNavigate } from 'react-router-dom';

const Signup = ({ signup }) => {

    const navigate = useNavigate();

    const[formData, setFormData] = useState({
        username:"",
        password:"",
        firstName:"",
        lastName:"",
        email:""
    });

    

    const handleChange = (evt) => {
        const {name, value} = evt.target;
        setFormData(prevFormData => ({
            ...prevFormData, [name]:value
        }));
    }

    const handleSubmit = async(evt) => {
        evt.preventDefault();

        try {
            await signup(formData);
            navigate('/');  //Redirects to homepage if sign up success.
        } catch (error) {
            console.error("Sign up failed:" , error);
        }

    }

    return(
        <div>
            
            <form onSubmit={handleSubmit}>
                <h3>Sign Up</h3>
                <div>
                    <label>Username</label>
                    <input
                    name="username"
                    type="text"
                    value={formData.username}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Password</label>
                    <input
                    name="password"
                    type="password"
                    value={formData.password}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>First Name</label>
                    <input
                    name="firstName"
                    type="text"
                    value={formData.firstName}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Last Name</label>
                    <input
                    name="lastName"
                    type="text"
                    value={formData.lastName}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Email</label>
                    <input
                    name="email"
                    type="email"
                    value={formData.email}
                    onChange={handleChange}
                    required
                    />
                </div>

                <button type="submit">Submit</button>

            </form>
        </div>
    )
}

export default Signup;