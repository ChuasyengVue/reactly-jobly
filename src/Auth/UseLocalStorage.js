// import { useState } from "react";

// function useLocalStorage(key, initialValue) {
//     const [storedValue, setStoredValue] = useState(() => {
//         try {
//             const item = window.localStorage.getItem(key);
//             return item ? JSON.parse(item) : initialValue;
//         } catch (error) {
//             console.error(error);
//             return initialValue;
//         }
//     });

//     const setValue = (value) => {
//         try {
//             const valueToStore = value instanceof Function ? value(storedValue) : value;
//             setStoredValue(valueToStore);
//             window.localStorage.setItem(key, JSON.stringify(valueToStore));
//         } catch (error) {
//             console.error(error);
//         }
//     };

//     return [storedValue, setValue];
// }

// export default useLocalStorage;  // Make sure this is a default export


import { useState, useEffect } from "react";

/** Custom hook for keeping state data synced with localStorage.
 *
 * This creates `item` as state and look in localStorage for current value
 * (if not found, defaults to `firstValue`)
 *
 * When `item` changes, effect re-runs:
 * - if new state is null, removes from localStorage
 * - else, updates localStorage
 *
 * To the component, this just acts like state that is also synced to/from
 * localStorage::
 *
 *   const [myThing, setMyThing] = useLocalStorage("myThing")
 */

function useLocalStorage(key, firstValue = null) {
  const initialValue = localStorage.getItem(key) || firstValue;

  const [item, setItem] = useState(initialValue);

  useEffect(function setKeyInLocalStorage() {
    console.debug("hooks useLocalStorage useEffect", "item=", item);

    if (item === null) {
      localStorage.removeItem(key);
    } else {
      localStorage.setItem(key, item);
    }
  }, [key, item]);

  return [item, setItem];
}

export default useLocalStorage;
