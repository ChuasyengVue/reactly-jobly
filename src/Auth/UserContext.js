import  { createContext, useState } from 'react';

export const UserContext = createContext({
    currentUser: null,
    login: () => {},
    signup: () => {},
    logout: () => {},
});

export const UserProvider = ({ children }) => {
    const [currentUser, setCurrentUser] = useState(null);

    const updateUser = (updatedUser) => {
        setCurrentUser((prevUser) => ({
            ...prevUser,
            ...updateUser,
        }));
    }

    return(
        <UserContext.Provider value={{currentUser, updateUser }}>
            {children}
        </UserContext.Provider>
    )
}