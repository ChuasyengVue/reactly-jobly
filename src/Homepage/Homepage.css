.homepage {
    display: flex; /* Enable flexbox */
    flex-direction: column; /* Stack items vertically */
    justify-content: center; /* Center vertically */
    align-items: center; /* Center horizontally */
    height: 100vh; /* Full viewport height */
    text-align: center; /* Center text */

}

.text {
    color: white; /* Set header color to white */
}
