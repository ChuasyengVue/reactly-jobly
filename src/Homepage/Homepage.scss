@use 'sass:math';

$w: 4em;
$a: 60deg;
$h: $w*math.tan(90deg - .5*$a);

html {
	--s: from #{.5*$a},;
	--l: 0% #{180deg - $a}, #0000 0% 50%;
	background:
		repeating-conic-gradient(var(--s) #b65333 var(--l)), 
		repeating-conic-gradient(var(--s) #f4bb3a var(--l)) 0 $h, 
		repeating-conic-gradient(var(--s) #ed7b32 var(--l)) 0 2*$h, 
		repeating-conic-gradient(var(--s) #966459 var(--l)) 0 3*$h, 
		linear-gradient(#f19b36 25%, #824739 0 50%, #fadc9c 0 75%, #d6552b 0);
	background-size: $w 4*$h
}