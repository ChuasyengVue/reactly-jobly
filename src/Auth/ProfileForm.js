import React, { useState, useContext, useEffect } from "react";
import JoblyApi from "../Api/Api";
import { UserContext } from "./UserContext";


const ProfileForm = () => {
  
    const {currentUser, updateUser} = useContext(UserContext);

  const [formData, setFormData] = useState({
    username: currentUser.username,
    firstName: currentUser.firstName,
    lastName: currentUser.lastName,
    email: currentUser.email,
    password: "",
  });

  useEffect(() => {
    if(currentUser) {
        setFormData({
            
            firstName:currentUser.firstName || '',
            lastName:currentUser.lastName || '',
            email:currentUser.email || '',
            password:currentUser.password || ''
        });
    }
  },[currentUser]);

  async function handleSubmit(evt) {
    evt.preventDefault();
    try {
      const updatedUser = await JoblyApi.updateProfile(currentUser.username, formData); // Update user profile
      updateUser(updatedUser);
      alert("Profile updated successfully!");
    } catch (err) {
      console.error("Profile update failed", err);
      
    }
  }

  async function handleChange(evt) {

    const {name, value} = evt.target;
    setFormData(formData => ({...formData, [name]:value}));
  }

  return (
    <form onSubmit={handleSubmit}>
    <div>
      <label>Username</label>
      <input name="username" value={formData.username} disabled />
      </div>
      <div>
      <label>First Name</label>
      <input
        name="firstName"
        value={formData.firstName}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Last Name</label>
      <input
        name="lastName"
        value={formData.lastName}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Email</label>
      <input
        name="email"
        value={formData.email}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Password</label>
      <input
        name="password"
        type="password"
        value={formData.password}
        onChange={handleChange}
      />
      </div>

      <button type="submit">Save Changes</button>
    </form>
  );
};

export default ProfileForm;
