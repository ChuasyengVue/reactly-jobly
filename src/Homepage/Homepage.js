import React, { useContext } from "react";
import { Link } from "react-router-dom";
import {UserContext} from "../Auth/UserContext";
import './Homepage.scss';
import './Homepage.css';

const Homepage = () => {

    const { currentUser } = useContext(UserContext);

    return(
        <div className="homepage">
            <h1 className="text">Jobly</h1>
            <p className="text"> All the jobs in one, convenient place.</p>
            {currentUser ? (
                <h2>Welcome Back, { currentUser.username }!</h2>
            ) : (
                <p>
                     <Link className="btn btn-primary font-weight-bold mr-3" to='/login'>Login</Link>
                     <Link className="btn btn-primary font-weight-bold" to='/signup'>Sign Up</Link>
                </p>
            )}
        </div>
    )
}

export default Homepage;