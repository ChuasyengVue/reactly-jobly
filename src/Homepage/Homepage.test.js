import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../Auth/UserContext";

//smoke test
it('renders without crashing', function () {
    render(<Homepage />);
});

//snapshot
it("matches snapshot", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider  currentUser={null}>
                <Homepage />
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

