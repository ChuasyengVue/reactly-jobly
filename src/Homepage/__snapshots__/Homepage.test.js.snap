// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <h1
      class="text"
    >
      Jobly
    </h1>
    <p
      class="text"
    >
       All the jobs in one, convenient place.
    </p>
    <p>
      <a
        class="btn btn-primary font-weight-bold mr-3"
        href="/login"
      >
        Login
      </a>
      <a
        class="btn btn-primary font-weight-bold"
        href="/signup"
      >
        Sign Up
      </a>
    </p>
  </div>
</DocumentFragment>
`;
